Source/WebCore/PAL/pal/text/TextCodec.cpp

```diff
@@ -49,14 +49,20 @@ int TextCodec::getUnencodableReplacement(char32_t codePoint, UnencodableHandling
         codePoint = replacementCharacter;
 
     switch (handling) {
-    case UnencodableHandling::Entities:
-        return snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "&#%u;", static_cast<unsigned>(codePoint));
-    case UnencodableHandling::URLEncodedEntities:
-        return snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "%%26%%23%u%%3B", static_cast<unsigned>(codePoint));
+    case UnencodableHandling::Entities: {
+        int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "&#%u;", static_cast<unsigned>(codePoint));
+        ASSERT(count >= 0);
```

Comment:
Seems a little excessive to have both this assertion and the `std::max(0, count)` below.

Reply:
If there is an unknown bug where `snprintf()` is returning a negative value, I want that `ASSERT()` to crash in Debug builds so I can find and fix the bug sooner.  If I remove the `ASSERT()`, it's a lot easier to miss that the `snprintf()` function was returning a negative value because the code is designed to handle that case without crashing or introducing a security issue in Release/Production builds.